crates/db_schema/src/impls/federation_queue_state.rs

```diff
@@ -0,0 +1,46 @@
+use crate::{
+  newtypes::{ActivityId, InstanceId},
+  source::federation_queue_state::FederationQueueState,
+  utils::{get_conn, DbPool},
+};
+use chrono::{TimeZone, Utc};
+use diesel::{prelude::*, result::Error};
+use diesel_async::RunQueryDsl;
+
+impl FederationQueueState {
+  /// load state or return a default empty value
+  pub async fn load(
+    pool: &mut DbPool<'_>,
+    instance_id_: InstanceId,
+  ) -> Result<FederationQueueState, Error> {
+    use crate::schema::federation_queue_state::dsl::{federation_queue_state, instance_id};
+    let conn = &mut get_conn(pool).await?;
+    Ok(
+      federation_queue_state
+        .filter(instance_id.eq(&instance_id_))
+        .select(FederationQueueState::as_select())
+        .get_result(conn)
+        .await
+        .optional()?
+        .unwrap_or(FederationQueueState {
+          instance_id: instance_id_,
+          fail_count: 0,
+          last_retry: Utc.timestamp_nanos(0),
+          last_successful_id: ActivityId(-1), // this value is set to the most current id for new instances
```

Comment:
Would be cleaner to make these fields optional/nullable imo.

Reply:
Yeah, that's true. Not 100% sure why I did it like this, probably because I'm using this struct both for the in-memory state as well as the DB state, and these values are almost always going to be non-optional so checking for presence everywhere is a bit of a pain.

This was part of the original implementation, I just moved it to a different file here. Let me know if you want me to change it, it would need a DB migration too.

Reply:
Maybe it would be possible to use `Lazy` or `OnceCell`?

Reply:
I've changed them to be optional instead of using "fake" values